src/Button/types.ts

```diff
@@ -13,7 +13,10 @@ export type VariantType = 'default' | 'primary' | 'invisible' | 'danger' | 'outl
 
 export type Size = 'small' | 'medium' | 'large'
 
-type StyledButtonProps = ComponentPropsWithRef<typeof StyledButton>
+/**
+ * Remove styled-components polymorphic as prop, which conflicts with radix's
+ */
+type StyledButtonProps = Omit<ComponentPropsWithRef<typeof StyledButton>, 'as'>
```

Comment:
the crux of the fix here seems like the `as` prop applied by styled-components differs from the types of that applied using the Radix polymorphic helper types an an incompatible way.  By omitting the styled component version of this type from the types that get passed around, we can rely on the radix version and avoid weird issues when trying to apply the as prop to the button base to the left of the spread restProps